### PR TITLE
Adding OMS Proxy client

### PR DESCRIPTION
**Pull request by fabianaferreira**
#### What is the purpose of this pull request?
- Adds the OMS Proxy client
- Improvements on `README.md`
- Changes some metric names for tracing
- Adds reference to Github package repository (@tiagonapoli's suggestion)

#### How should this be manually tested?
Link in a project such as `service-example`

#### Types of changes
- [ ] Refactor (non-breaking change that only makes the code better)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

#### Chores checklist
- [x] Update `CHANGELOG.md`

### CODE REVIEW

File: src/clients/omsApiProxy.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  public register(body: CustomDataRegistry): Promise<string> {
+    return this.http.post('register', body, {
+      metric: 'registerCustomData-post',
+    })
+  }
+}
+
+interface CustomData {
```
Comment (lucis) [inline src/clients/omsApiProxy.ts:44]: Custom Data is just these fields?

File: src/clients/omsApiProxy.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ * },
+ *
+ */
+export class OMSProxy extends AppClient {
+  constructor(context: IOContext, options?: InstanceOptions) {
+    super('vtex.oms-api-proxy@0.x', context, options)
+  }
+
+  public orders(id: string): Promise<string> {
+    return this.http.get(`/orders/${id}`, {
```
Comment (lucis) [inline src/clients/omsApiProxy.ts]:
```suggestion
    return this.http.get<Order>(`/orders/${id}`, {
```

Don't we have the returning types for all of these methods?

File: README.md [hunk 3]
```diff
@@ -13,29 +13,32 @@ This exports **Clients** and Typescript **typings** to help you connecting a VTE
 | Catalog         | `getSkuById`                                                |
 | Logistics       | `getDockById`, `pickupById`, `nearPickupPoints`, `shipping` |
 | OMS             | `userLastOrder`, `order`                                    |
+| Checkout        | `getOrderFormConfiguration`, `setOrderFormConfiguration`, `setSingleCustomData` |
+| OMS Proxy       | `orders`, `orderFormId`, `customData`, `register`           |
+---
 
-
+In order to know a little bit more about the methods that the `OMS Proxy` client implements, you can check out [this document](https://www.notion.so/How-to-use-the-OMS-API-Proxy-application-e82f11ff896247c58a7e2e658d631516). It's important to emphasize that to use this client, you will have to declare a dependency and a policy, this is also describe in the document mentioned above.
```
Comment (lucis) [inline README.md]: I'd suggest adding a column on the table above and linking this resource there. If we add some other clients, both data will be apart on the document.

I think we can change this later with a proposal for documentation (I'm thinking a table for each client with a reserved section)

**Review by lucis — APPROVED** [review]

File: src/clients/omsApiProxy.ts (hunk 1 above)
Comment (fabianaferreira) [reply src/clients/omsApiProxy.ts:44]: To insert it, yes. However, this method is restricted to custom data of only one value. The `appId` that is necessary is got from the headers, on the `oms-api-proxy`.
[Reference](https://developers.vtex.com/reference/custom-data#setsinglecustomfieldvalue)

**Review by lucis — APPROVED** [review]
Really nice

**Pull request merged**